GLOSSARY.md

```diff
@@ -0,0 +1,40 @@
+# GitOps Glossary
+
+This glossary accompanies the [GitOps Principles](./PRINCIPLES.md), and other supporting documents in this repository.
+
+- ## Continuous
+
+    "Continuous" is intended to match the industry standard term: [reconciliation](#reconciliation) continues to happen, not that it must be instantaneous.
+
+- ## Declarative Description
+
+    A configuration that describes the desired operating state of a system without specifying procedures for how that state will be achieved. This separates configuration (the desired state) from the implementation (commands, API calls, scripts etc.) used to achieve that state.
+
+- ## Desired State
+
+    The aggregate of all configuration data needed to start, bootstrap, recreate, run, and recover a system. This does not include data stored within a system, eg. database contents.
```

Comment:
Are you sure you want to exclude data stored within the system -- on the face of it, this would rule out self-describing systems like Kubernetes, in which the data describing the system is also data within the system (or require you to have a tiered classification, such that e.g., custom resources are data on one level, but configuration on another level and thus OK for GitOps on that level ...)

Reply:
The intention is here is basically not to say that every change to a database has to go through git, or that a 50 GB database somehow needs to be stored in git. If we have CRDs or configmaps that are needed to start, bootstrap, recreate, and recover a system then they would need to be stored along with the rest of the config. 

Does that make sense or do you think it needs additional clarification?

Reply:
```suggestion
    The aggregate of all configuration data needed to start, bootstrap, recreate, run, and recover a system. Optionally, this can include application data stored within applications in a system, eg. application database contents.
```
@squaremo @todaywasawesome WDYT?
• I added `application` modifier to classify application data differently from system data.
• [Application software](https://en.wikipedia.org/wiki/Application_software) is typically different from [system software](https://en.wikipedia.org/wiki/System_software).
• Then, I added `Optionally,` to give people the autonomy to include application data.
• For example, [Liquibase](https://www.liquibase.org/) can be used to include 0.1 GB application database schema via GitOps, while the remaining 49.9 GB of data are excluded from GitOps.
• The right shade of gray is situational.

Reply:
Yeah, I get the intention, and it's tricky to nail it down without tying yourself in logical knots. Probably I'm being pedantic.

On another point though:

> The aggregate of all configuration data needed to start, bootstrap, recreate, run, and recover a system.

I don't think you want to specify "all configuration data needed to [...] run [...] a system". I think the intention there is to account for the desired state being a target while the system is running. But taken at face value, it draws in the bookkeeping the system does: the desired state is usually abstract, while the state needed to _run_ the system must have all the details. For example, you might have a Kubernetes Deployment in your declarations, but the running system needs all the pods to be defined. There is a large belt of grey area, extending into databases (as in my comment above) but also extending the other direction into configuration.

The approach of relating the description to the system's requirements is the right one, I think. Perhaps something like

> An  abstract description of the system that is sufficient to create and recover the running system with automation, reliably.

"Reliably" is doing a lot of work there -- what I want to get across is that there's a tolerance for details changing.

I don't mention bootstrapping because I consider that part of the tooling, rather than the system. I don't mention running, because I think a description that's adequate for recovering the system must also be adequate for keeping it running. (But that point might be worth spelling out)

Reply:
@lloydchang I think I understand what you mean.

- Starting off with "generally this does not include" was ambiguous, it sounds like you took it to mean gitops does not include this at all. Changing that to "This configuration data generally does not include" to make that more clear. Now does not exclude these less common use cases you mention. 
- Also the sentence continued with "but may include credentials…". The "but may" here seems to imply the application data "may not" be included. Changing that to "though often does include".
- Per @todaywasawesome changing "and configuration for recovery tools that should be running on the system" to "or configuration for data recovery tools running on that system", since these are two of the most common, but still optional ways of doing stateful applications with GitOps safely.

```diff
## Desired State

- This generally does not include persistent application data, eg. database contents, but may include credentials for accessing that data, and configuration for recovery tools that should be running on the system.
+ This configuration data generally does not include persistent application data, eg. database contents, though often does include credentials for accessing that data, or configuration for data recovery tools running on that system.

```

Reply:
@squaremo agreed on both points. Incorporating a version that we think satisfies all these points. 

Everyone, please lmk if you still feel it should change after this ebdc34bc2d5448770260325e9d243b85dbe27641 🙏 😸